docs/examples/nmt/gnmt.md

```diff
@@ -0,0 +1,456 @@
+# Google NMT on IWSLT 2015 English-Vietnamese Translation
+
+In this notebook, we are going to train Google NMT on IWSLT 2015 English-Vietnamese
+Dataset. The building prcoess includes four steps: 1) load and process dataset, 2)
+create sampler and DataLoader, 3) build model, and 4) write training epochs.
+
+## Load MXNET and Gluon
+
+```python
+import argparse
+import time
+import random
+import os
+import io
+import logging
+import numpy as np
+import mxnet as mx
+from mxnet import gluon
+from mxnet.gluon.data import ArrayDataset, SimpleDataset
+from mxnet.gluon.data import DataLoader
+import gluonnlp.data.batchify as btf
+from gluonnlp.data import ExpWidthBucket, FixedBucketSampler, IWSLT2015
+from gluonnlp.model import BeamSearchScorer
+from scripts.nmt.gnmt import get_gnmt_encoder_decoder
+from scripts.nmt.translation import NMTModel, BeamSearchTranslator
+from scripts.nmt.loss import SoftmaxCEMaskedLoss
+from scripts.nmt.utils import logging_config
+from scripts.nmt.bleu import compute_bleu
+import scripts.nmt._constants as _C
+```
+
+## Hyper-parameters
+
+```python
+np.random.seed(100)
+random.seed(100)
+mx.random.seed(10000)
+ctx = mx.gpu(0)
+
+# parameters for dataset
+dataset = 'IWSLT2015'
+src_lang = 'en'
+tgt_lang = 'vi'
+src_max_len = 50
+tgt_max_len = 50
```

Comment:
for related hyperparameters, we can do them on the same line, such as `src_max_len, tgt_max_len = 50, 50`